Adding slightly tested support for OGG, using VorbisComment parsing from flac.go

Simon-L opened the pull request:
As the commit message states, this is slightly tested, I just grabbed a few ogg albums from various sources to test against, it works.
It took me a few hours to write this code from reading the Ogg documentation and opening .ogg files in an hex editor so it might not be failproof.
I should add I'm relatively new to Go so please make any change you see fit to make the code look better, it's relatively verbose as is! :)

Since the metadata format is the same, it's using the `readVorbisComment()` function from the flac.go file.
As a result it returns `metadataFLAC` and returns `FLAC` as its format.
Maybe we could change the FLAC format name to `"Vorbis"` since they are fundamentaly the same in terms of metadata encoding.

dhowden commented:
Thanks for the patch - looks great!  I have managed to find some OGG files to test it on, so I'm running through those.  I haven't looked at reading OGG files - so this is new to me!  I will have a look at the standard.

I've added a few comments to the code so if you could look at that it would be awesome (they are all about adding some code-comments to explain the `Seek` calls, which would make it much easier to read and link back to the OGG format).

Simon-L commented:
I added some comments and links to the vorbis specification.
It's not using a type like `BlockType` in `flac.go`, instead I added constants for the packet types.
It might use some change in the future just for the sake of having unified code.

dhowden commented:
Thanks, that's great.  If you will rebase the commits into one then I'll merge it in.

dhowden commented:
Merged as e5e85d7

Thanks for the patch!
